tests/subsys/mgmt/mcumgr/mcumgr_client/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(mcumgr_client)
+
+FILE(GLOB app_sources
+	src/*.c
+)
+
+target_sources(app PRIVATE ${app_sources})
+target_sources(app PRIVATE ${ZEPHYR_BASE}/subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c)
+target_sources(app PRIVATE ${ZEPHYR_BASE}/subsys/mgmt/mcumgr/grp/os_mgmt_client/src/os_mgmt_client.c)
+
+add_compile_definitions(CONFIG_SMP_CLIENT=y)
+add_compile_definitions(CONFIG_SMP_CMD_DEFAULT_LIFE_TIME=5)
+add_compile_definitions(CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER=1)
+add_compile_definitions(CONFIG_MCUMGR_GRP_OS_ECHO_REQ=y)
```

Comment:
For testing like this, I think this should be designed around the "unit testing" board which excludes most zephyr files - @tejlmand thoughts?

Reply:
I update config that prj.conf enable main setup. I will stub SMP client for unit test for testing MCUmgr client component's